textpipe/doc.py

```diff
@@ -413,3 +414,84 @@ def similarity(self, other_doc, metric='jaccard', hash_method='minhash'):
                                       f'/{hash_method} is not implemented as similarity metric')
+
+    @property
+    def word_vectors(self):
+        """
+        Returns word embeddings for the words in the document.
+        The used spacy models don't have "true" word vectors
+        but only context-sensitive tensors that are within the document.
+
+        Returns:
+        A dictionary mapping words from the document to a dict with the
+        corresponding values of the following variables:
+
+        has vector: Does the token have a vector representation?
+        vector norm: The L2 norm of the token's vector (the square root of the
+                    sum of the values squared)
+        OOV: Out-of-vocabulary (This variable always gets the value True since
+                                there are no vectors included in the model)
+        vector: The vector representation of the word
+
+        >>> doc = Doc('Test sentence')
+        >>> doc.word_vectors['Test']['is_oov']
+        True
+        >>> doc.word_vectors['Test']['vector'].shape[-1]
+        384
+        >>> doc.word_vectors['Test']['vector_norm'] == doc.word_vectors['sentence']['vector_norm']
+        False
+        """
+        return {token.text: {'has_vector': token.has_vector,
+                             'vector_norm': token.vector_norm,
+                             'is_oov': token.is_oov,
+                             'vector': token.vector}
+                for token in self._spacy_doc}
+
+    @property
+    def doc_vector(self):
+        """
+        Returns document embeddings based on the words in the document.
+
+        >>> import numpy
+        >>> numpy.array_equiv(Doc('a b').doc_vector, Doc('a b').doc_vector)
+        True
+        >>> numpy.array_equiv(Doc('a b').doc_vector, Doc('a a b').doc_vector)
+        False
+        """
+        return self.aggregate_word_vectors()
+
+    def aggregate_word_vectors(self, aggregation='mean', normalize=False, exclude_oov=False):
+        """
+        Returns document embeddings based on the words in the document.
+
+        >>> import numpy
+        >>> doc1 = Doc('a b')
+        >>> doc2 = Doc('a a b')
+        >>> numpy.array_equiv(doc1.aggregate_word_vectors(), doc1.aggregate_word_vectors())
+        True
+        >>> numpy.array_equiv(doc1.aggregate_word_vectors(), doc2.aggregate_word_vectors())
+        False
+        >>> numpy.array_equiv(doc1.aggregate_word_vectors(aggregation='mean'), doc2.aggregate_word_vectors(aggregation='sum'))
+        False
+        >>> doc = Doc('sentence with an out of vector word lsseofn')
+        >>> doc.aggregate_word_vectors().shape
+        (384,)
+        >>> numpy.array_equiv(doc.aggregate_word_vectors(exclude_oov=False), doc.aggregate_word_vectors(exclude_oov=True))
+        False
+        """
+        if exclude_oov:
+            tokens = [token for token in self._spacy_doc if not token.is_oov]
+        else:
+            tokens = [token for token in self._spacy_doc]
```

Comment:
the above 4 lines could be 
```suggestion
            tokens = [token for token in self._spacy_doc if not exclude_oov or not token.is_oov]
```

Reply:
I agree that's indeed nicer